src/BankHolidayResolver.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * @copyright 2017 Hostnet B.V.
+ */
+declare(strict_types=1);
+
+namespace Hostnet\Component\BankHoliday;
+
+/**
+ * Find bank holidays based on date or type.
+ * @see BankHoliday
+ */
+class BankHolidayResolver
+{
+    /**
+     * @var string[]
+     */
+    private $available_bank_holidays = [];
+
+    /**
+     * @param string[] $available_bank_holidays
+     */
+    public function __construct(array $available_bank_holidays = BankHoliday::ALL)
+    {
+        foreach ($available_bank_holidays as $available_bank_holiday) {
+            $this->addAvailableBankHoliday($available_bank_holiday);
+        }
+    }
+
+    private function addAvailableBankHoliday(string $bank_holiday): self
+    {
+        if (!in_array($bank_holiday, BankHoliday::ALL)) {
+            throw new \RuntimeException(sprintf(
+                "Tried to add unknown BankHoliday '%s'. Only add BankHolidays from %s.",
+                $bank_holiday,
+                BankHoliday::class
+            ));
+        }
+
+        if (!in_array($bank_holiday, $this->available_bank_holidays)) {
+            $this->available_bank_holidays[] = $bank_holiday;
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getAvailableBankHolidays(): array
+    {
+        return $this->available_bank_holidays;
+    }
+
+    /**
+     * @param \DateTime $date
+     * @return string[]
+     */
+    public function getByDate(\DateTime $date): array
+    {
+        $date->setTime(0, 0, 0);
```

Comment:
clone the date to prevent side effects.

```php
$date = new \DateTime('2017-10-10 11:10:09);
$a = new BankHolidayResolver();

var_dump($date->format('c')); // string(25) "2017-10-10T11:10:09+00:00"

$a->getByDate($date);

var_dump($date->format('c')); // string(25) "2017-10-10T00:00:00+00:00"
```